docs/source/concept_models/vivarium_census_synthdata/concept_model.rst

```diff
@@ -321,6 +321,16 @@ from ACS:
 | 38    | Noninstitutionalized group quarters population   |
 +-------+--------------------------------------------------+
 
+Vivarium needs to know the population size that it will initialize
+before a simulation starts running. Because we would like to set a
+number of households and then sample them, we don't have a total
+population count to give to Vivarium.  As a work-around solution we
```

Comment:
The way I ended up implementing this was slightly different: we sample households until we exceed the target population size `N`, then throw out the last household such that we are at `m<=N`. We then add back in `N-m` untracked people. This gets us closer to `N` than immediately stopping once we've exceeded `N-17`.

Reply:
edit made, but I just started wondering if 17 is the largest in the ACS data you are using, or in any ACS data.

Reply:
it's just the largest household in the data that i am using -- i'm not sure if other states or years have larger households. but i no longer have 17 hard-coded per the method explained above

maybe:


Vivarium needs to know the population size that it will initialize before a simulation starts running. Because we would like to set a number of households and then sample them, we don't have a total population count to give to Vivarium.  As a work-around solution we will fix a target number of people, sample households until we exceed this number, then remove the household that pushed us above our target. If this results in fewer than the target number of simulants, we will add back in untracked simulants to meet our target. The handful of untracked sims are not very much extra memory to drag through the full sim. A drawback is that this leaves both the number of households and the number of simulants as variable (but not that variable).

(i don't know if you had already written something that's saved elsewhere? i wasn't seeing the update)

Reply:
oh oops nevermind i see it now --- i was looking somehow at the wrong view. please disregard!